controllers/StockController.php

```diff
@@ -515,6 +515,53 @@ public function Stockentries(\Psr\Http\Message\ServerRequestInterface $request,
 		]);
 	}
 
+	public function StockMetricsPurchases(\Psr\Http\Message\ServerRequestInterface $request, \Psr\Http\Message\ResponseInterface $response, array $args)
+	{
+		if (isset($request->getQueryParams()['start_date']) AND isset($request->getQueryParams()['end_date']))
+		{
+			$start_date = $request->getQueryParams()['start_date'];
+			$end_date = $request->getQueryParams()['end_date'];
+			$where = "purchased_date >= '$start_date' AND purchased_date <= '$end_date'";
+		}
+		else
+		{
+			// Default this month
+			$where = "purchased_date >= DATE(DATE('now', 'localtime'), 'start of month')";
+		}
+
+
+		if (isset($request->getQueryParams()['byGroup']))
+		{
+			$sql = "
+			SELECT product_group_id as id, product_group as name, sum(quantity * price) as total
+			FROM product_purchase_history
+			where $where
+			GROUP BY product_group
+			ORDER BY product_group
+			";
+		} else {
```

Comment:
Would be great if the current code style could be used, e.g. new lines after curly braces, camel case variable names and so on ([ref for PHP code](https://github.com/grocy/grocy/blob/master/.php-cs-fixer.php)) - but I can also take care of that whenever merging that.

Reply:
I agree. I'll review and adjust the style to the preferred standard.

Reply:
Fixed :heavy_check_mark: